File: backup/count-grams.py
import fileinput
import operator
from collections import defaultdict

counts = defaultdict(int)

def p((k,v)):
	print k + '\t' + str(v) 

for line in fileinput.input():
	tokens = line.split('\t')
	actions = '\t'.join(tokens[:-1])
	counts[actions] += 1

counts = sorted(counts.iteritems(), key=operator.itemgetter(1), reverse= True)
counts = filter(lambda (k,v): True if v >1 else False, counts)
map(p, counts)

File: backup/count-grams.rb
counts = Hash.new(0)

ARGF.each_line do |line|
  tokens = line.split("\t")
  counts[tokens] += 1
end

#todo dynamic output instead of static output
#todo some kind of string similarity measure instead of having them be exactly the same
counts.sort_by{|k,v| v*-1 }.each do |el|
  if el[1] >= 1
    puts "#{el[0].join("\t").strip}\t#{el[1]}"
  end
end